### PR TITLE
Dpwebcore 892/add coverage

### PR DESCRIPTION
**Pull request by benwainwright**
@bbc/web-core-devx

# Problem

-   We agreed on a coverage threshold but we currently have no testing or coverage reporting in place
-   Wider context: https://jira.dev.bbc.co.uk/browse/DPWEBCORE-892

# Solution

- Add tests to meet 85% threshold
- Configure jest to fail if threshold is not met 

## Notes

- We agreed to not enforce the threshold for infrastructure code; but since we are currently *meeting* it, I've not added the configuration yet. If this gets harder; we can add that exclusion.
- I've removed the service worker as it doesn't seem necessary and I'd have to add a bunch of tests to cover it
- I've added test reporting for both tests and coverage (recently added CodeBuild feature)
- Files contained within `./src` (rather than its subfolders) are excluded. This should only contain entry point files that are impossible to test, and they should delegate as much as possible to testable functions

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -63,6 +67,26 @@
...
     ]
   },
+  "jest": {
+    "collectCoverageFrom": [
+      "./src/**/*.{js,jsx,ts,tsx}",
+      "!./**/*.d.ts",
+      "!./src/*"
+    ],
+    "coverageReporters": [
+      "text",
```
Comment (jameshd) [inline package.json:77]: could we add the `lcov`reporter please - I find it's probably the most useful 👍 

https://jestjs.io/docs/en/configuration#coveragereporters-arraystring--string-options

File: src/app/render.test.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import render from "./render";
+
+describe("Render", () => {
+  it("Renders the app without crashing", () => {
```
Comment (abrarkhan1234) [inline src/app/render.test.js:4]: Can this test be renamed to "Renders the app succesfully"? And then assert text from the app exists?

File: .gitignore [hunk 3]
```diff
@@ -1,2 +1,6 @@
 template
-build
+build/*
+src/react-app-env.d.ts
+coverage
+junit.xml
+yarn-error.log
```
Comment (abrarkhan1234) [inline .gitignore:6]: can we add `node_modules` here?

**Comment by abrarkhan1234** [comment]
Are we doing too much in this PR, there's infrastructure stuff as well as app refactor?

File: package.json (hunk 1 above)
Comment (benwainwright) [reply package.json:77]: Done

File: src/app/render.test.js (hunk 2 above)
Comment (benwainwright) [reply src/app/render.test.js:4]: > Can this test be renamed to "Renders the app successfully"?

Sure, I'll do that.

> And then assert text from the app exists?

* That's an entirely different test. If we *did* add the one you are suggesting, I'd still want to keep the test I've written here in its current format.
* I really don't see the value in asserting on the contents of static strings

File: .gitignore (hunk 3 above)
Comment (benwainwright) [reply .gitignore:6]: Sure :)

**Comment by benwainwright** [comment]
> Are we doing too much in this PR, there's infrastructure stuff as well as app refactor?

@abrarkhan1234 Possibly. But I don't want to spend too much time on this ticket. The infrastructure stuff is also just a refactor.

**Pull request merged**